src/api/events.rs

```diff
@@ -18,10 +23,60 @@ pub type Events<'a> =
 #[allow(unused)]
 pub type InBlockEvents<'client> = TransactionEvents<'client, GearConfig, Event>;
 
+/// Subxt Error
+pub type SubxtError = subxt::Error<DispatchError>;
+
 impl Api {
     /// Subscribe all events
     #[allow(unused)]
     pub async fn events(&self) -> Result<Events<'_>> {
         Ok(self.runtime.events().subscribe().await?)
     }
+
+    /// Capture the dispatch info of any extrinsic and display the weight spent
+    pub async fn capture_dispatch_info<'e>(
+        &self,
+        tx: &TransactionInBlock<'e, GearConfig, DispatchError, Event>,
+    ) -> core::result::Result<InBlockEvents<'e>, SubxtError> {
```

Comment:
If you define custom `Result` with the default generic, you don't need to distinguish it with the core `Result`.

```rust
pub type Result<T, E = Error> = core::result::Result<T, E>;
```

```suggestion
    ) -> Result<InBlockEvents<'e>, SubxtError> {
```

Reply:
The reason is the errors inside this function could be converted into `subxt:Error` directly, but not for my custom Error